### PR TITLE
Add user interactions collection to the jssandbox

### PR DESCRIPTION
**Pull request by imsickofmaps**

### CODE REVIEW

File: js_sandbox/lib/go-nike-ghr.js [hunk 1]
```diff
@@ -87,7 +99,15 @@ function GoNikeGHR() {
...
                 return choice.value;
-            }, question.question, choices);
+            }, question.question, choices, null,
+                        {
+                            on_enter: function() {
+                                var action = "MANDL;question;" + question.question;
+                                var p_log = self.interaction_log(action);
+                                return p_log;
+                            }
+                        }
```
Comment (smn) [inline js_sandbox/lib/go-nike-ghr.js]: mixing tabs & spaces?

File: js_sandbox/lib/go-nike-ghr.js [hunk 2]
```diff
@@ -127,12 +154,45 @@ function GoNikeGHR() {
...
         p.add_callback(function(result) {
-            var json = self.check_reply(result, url, 'GET', false);
+            var json = self.check_reply(result, url, 'GET', null, false);
+            return json;
+        });
+        return p;
+    };
+
+    self.crm_post = function(path, data) {
+        var url = im.config.crm_api_root + path;
```
Comment (smn) [inline js_sandbox/lib/go-nike-ghr.js:193]: Where does this `im` variable come from?

File: js_sandbox/lib/go-nike-ghr.js [hunk 3]
```diff
@@ -105,7 +125,14 @@ function GoNikeGHR() {
 
             return new ChoiceState(state_name, function(choice) {
                 return choice.value;
-            }, view.opinion, choices);
+            }, view.opinion, choices, null,
+                        {
+                            on_enter: function() {
+                                var action = "OPINIONS;viewed;" + view.opinion;
```
Comment (smn) [inline js_sandbox/lib/go-nike-ghr.js]: Why are you posting it in this manner? Why not just urlencode or json to save the decoding on the receiving end?

Comment (smn) [inline js_sandbox/lib/go-nike-ghr.js]: You could do:

``` javascript
var action = self.urlencode({type: 'OPINIONS', action: 'viewed', opinion: view.opinion});
return self.interaction_log(action);
```

File: js_sandbox/test/fixtures/userinteraction_wwnd.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+    "url": "http://ghr.preview.westerncapelabs.com/api/userinteraction/",
+    "method": "POST",
+    "content_type": "application/x-www-form-urlencoded",
+    "data": "action=WWND%3Bndabaga%3Bviewed&transport=ussd&msisdn=1234567",
+    "body": {}
+}
```
Comment (smn) [inline js_sandbox/test/fixtures/userinteraction_wwnd.json]: Your fixtures are all missing a newline.

File: js_sandbox/lib/go-nike-ghr.js (hunk 3 above)
Comment (imsickofmaps) [inline js_sandbox/lib/go-nike-ghr.js]: There's no decoding to be done on the receiving end. It's just getting dumped into one column. I've just added a bit of structure to these ones but some may be structure-less.

File: js_sandbox/test/fixtures/userinteraction_wwnd.json (hunk 4 above)
Comment (imsickofmaps) [inline js_sandbox/test/fixtures/userinteraction_wwnd.json]: Fixed all

File: js_sandbox/lib/go-nike-ghr.js (hunk 3 above)
Comment (smn) [inline js_sandbox/lib/go-nike-ghr.js]: Gut feel is that this is a bad idea. What if your `view.opinion` contains a semicolon? Do you really want to maintain your own encoding scheme when you've got both json & urlencode available and you're doing an HTTP POST (so both are easy to use).

Comment (imsickofmaps) [inline js_sandbox/lib/go-nike-ghr.js]: Are you saying I should POST an arbitrary set of form fields over to the Django endpoint and it should reconstruct them there and stick them into the column?

File: js_sandbox/lib/go-nike-ghr.js (hunk 2 above)
Comment (imsickofmaps) [inline js_sandbox/lib/go-nike-ghr.js:193]: @smn Can we merge this is in with this supposedly working code (according to travis) and I'll open an issue to fix all instances if we go to QA and it breaks?

**Comment by smn** [comment]
:+1:

**Pull request merged**